s3_scrubber/src/main.rs

```diff
@@ -258,11 +227,25 @@ async fn main() -> anyhow::Result<()> {
                 &cli,
                 bucket_config,
                 console_config,
-                *node_kind,
-                *depth,
-                *skip_validation,
+                node_kind,
+                depth,
+                skip_validation,
             )
             .await
         }
+        Command::ScanMetadata {} => match scan_metadata(bucket_config).await {
+            Err(e) => {
+                tracing::error!("Failed: {e}");
+                Err(e)
+            }
+            Ok(summary) => {
+                println!("{}", summary.summary_string());
+                if summary.is_fatal() {
+                    Err(anyhow::anyhow!("Fatal scrub errors detected"))
```

Comment:
You could use `anyhow::bail` here.

Reply:
I don't think bail adds anything useful here: it would be the same as writing `return Err(...` instead of just `Err(...`, but we're already in a block whose output is the return value of the function.

I avoid `return` where possible, because it makes it easier to transplant a block of code into some other location where we might want to do something with the error rather than jumping out of the function.